tests/ci/commit_status_helper.py

```diff
@@ -82,26 +82,48 @@ def post_labels(gh, pr_info, labels_names):
         pull_request.add_to_labels(label)
 
 
-def fail_simple_check(gh, pr_info, description):
-    if SKIP_SIMPLE_CHECK_LABEL in pr_info.labels:
-        return
+def fail_mergeable_check(gh, pr_info, description):
     commit = get_commit(gh, pr_info.sha)
     commit.create_status(
-        context="Simple Check",
+        context="Mergeable Check",
         description=description,
         state="failure",
         target_url=GITHUB_RUN_URL,
     )
 
 
-def create_simple_check(gh, pr_info):
+def reset_mergeable_check(gh, pr_info, description=""):
     commit = get_commit(gh, pr_info.sha)
-    for status in commit.get_statuses():
-        if "Simple Check" in status.context:
-            return
     commit.create_status(
-        context="Simple Check",
-        description="Skipped",
+        context="Mergeable Check",
+        description=description,
         state="success",
         target_url=GITHUB_RUN_URL,
     )
+
+
+def update_mergeable_check(gh, pr_info):
+    if SKIP_MERGEABLE_CHECK_LABEL in pr_info.labels:
+        return
+
+    required = [
+        "Fast test",
+        "Style Check",
+        "ClickHouse build check",
+        "ClickHouse special build check",
+    ]
+    commit = get_commit(gh, pr_info.sha)
+    checks = {
+        check["context"]: check["state"]
+        for check in filter(
+            lambda check: (check["context"] in required),
+            reversed(commit.get_statuses()),
```

Comment:
Did you check locally how many API requests it does under the hood? Are you sure it's given the time ordered?

Reply:
`commit.get_statuses()` is one API call
at least the same test should be time ordered I think

Reply:
It must depend on `per_page`. [1](https://pygithub.readthedocs.io/en/latest/github_objects/Commit.html#github.Commit.Commit.get_statuses) + [2](https://docs.github.com/en/rest/commits/statuses#get-the-combined-status-for-a-specific-reference)

so [default](https://api.github.com/repos/ClickHouse/ClickHouse/commits/5a34cf901043a237bf490948ede31e91ebb6d447/status) vs [100](https://api.github.com/repos/ClickHouse/ClickHouse/commits/5a34cf901043a237bf490948ede31e91ebb6d447/status?per_page=100)

`commit.get_statuses()` and `reversed(commit.get_statuses())` are different. One returns the generator, and the second finalizes the request.

According to the following simple test, the statuses are ordered by ID, so it should work fine for us :+1: 

```
curl https://api.github.com/repos/ClickHouse/ClickHouse/commits/5a34cf901043a237bf490948ede31e91ebb6d447/status?per_page=100| jq '.statuses[].id' | sha256sum
751ee1ab7b9c1afc70b02e3a295b62ed3f7731d6827e93b48fa12aefae7bdb22  -
curl https://api.github.com/repos/ClickHouse/ClickHouse/commits/5a34cf901043a237bf490948ede31e91ebb6d447/status?per_page=100| jq '.statuses[].id' | sort | sha256sum
751ee1ab7b9c1afc70b02e3a295b62ed3f7731d6827e93b48fa12aefae7bdb22  -
```